### PR TITLE
[netcore] Disable runtime cleanup on netcore, coreclr does not cleanup either. Also disable aborting background threads, its no longer needed.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Wow. this might fix the Alpine Linux problem too, a hang at shutdown, related to alternative libc.

**Comment by vargaz** [comment]
This is a netcore only change.

**Comment by jaykrell** [comment]
Understood. This might fix it for netcore.

**Comment by marek-safar** [comment]
@lewurm/@lambdageek could you please review

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -3722,6 +3722,11 @@ mono_thread_manage (void)
 		mono_thread_execute_interruption_void ();
 	}
 
+#ifndef ENABLE_NETCORE
```
Comment (filipnavara) [inline mono/metadata/threads.c]: Is there a reason to guard this with ENABLE_NETCORE instead of DISABLE_CLEANUP?

Comment (jaykrell) [reply mono/metadata/threads.c]: That name is too short, but it is a valid point -- to compose out of a finer grained set of names, vs. one omnibus one. The fine grained ifdef raises the question as to what combinations of things work or are tested. But that does not necessarily bother people -- witness autoconf.

Comment (filipnavara) [reply mono/metadata/threads.c]: DISABLE_CLEANUP is already used at the other place in this commit, so it made me wonder whether all the combinations are actually necessary.

Comment (vargaz) [reply mono/metadata/threads.c]: Doesn't matter too much, DISABLE_CLEANUP is a --enable-minimal feature which can be used by anyone, while not aborting threads is  a part of netcore.

**Pull request merged**